doc/introduction/chemistry.rst

```diff
@@ -21,38 +21,38 @@ Eigensolver (VQE) algorithm.
 Building the electronic Hamiltonian
 -----------------------------------
 
-The ``qchem`` module provides access to a driver function :func:`~.generate_hamiltonian`
+The ``qchem`` module provides access to a driver function :func:`~.molecular_hamiltonian`
 to generate the electronic Hamiltonian in a single call. For example,
 
 .. code-block:: python
 
-    h, nr_qubits = qml.qchem.generate_hamiltonian(
+    h, n_qubits = qml.qchem.molecular_hamiltonian(
         name='h2',
         geo_file='h2.xyz',
         charge=0,
-        multiplicity=1,
-        basis_set='sto-3g',
-        n_active_electrons=2,
-        n_active_orbitals=2
+        mult=1,
+        basis='sto-3g',
+        nact_els=2,
+        nact_orbs=2
     )
 
 where:
 
 * ``h`` is the qubit Hamiltonian of the molecule represented as a PennyLane Hamiltonian, and
 
-* ``nr_qubits`` is the number of qubits operators needed to represent it.
+* ``n_qubits`` is the number of qubits operators needed to represent it.
 
-Internally, :func:`~.generate_hamiltonian` calls the following functions in order
+Internally, :func:`~.molecular_hamiltonian` calls the following functions in order
 to generate the qubit Hamiltonian:
 
 .. currentmodule:: pennylane_qchem.qchem
 
 .. autosummary::
 
     read_structure
-    meanfield_data
+    meanfield
     active_space
-    decompose_hamiltonian
+    decompose_molecular_hamiltonian
```

Comment:
I would propose that we avoid using more than one underscore "_" in naming functions or variables. How about just `decompose`? Since the input is a Hamiltonian, it should be clear that we are decomposing the Hamiltonian

Reply:
@ixfoduap `decompose` does not tell much to me. I don't like too long names as well but this function is constrained to decompose the molecular Hamiltonian as it relies on OpenFermion `MolecularData` class. Also, this function is marked for **deprecation** after the generalization of the `observable` function to build general Hamiltonians. Let me know if you want to go with `decompose`.

Reply:
A user only has to learn what an argument means once, but has to type it every time :laughing: : 
It's ok if an argument name gives 90% of the intuition of what it does to the user, but is much easier to type repeatedly (the docstring is the place for fully explaining)

So in this case `decompose` does a good job (and the docstring can explain that it applies to that particular class)